test/program_test.go

```diff
@@ -157,8 +158,41 @@ var _ = Describe("Creating a YAML program", func() {
 			Expect(stack.Status.LastUpdate.State).To(Equal(shared.SucceededStackStateMessage))
 			Expect(apimeta.IsStatusConditionTrue(stack.Status.Conditions, pulumiv1.ReadyCondition)).To(BeTrue())
 		})
-	})
 
+		When("the program is changed", func() {
+			var revisionOnFirstSuccess string
+
+			BeforeEach(func() {
+				prog := programFromFile("./testdata/test-program.yaml")
+				Expect(k8sClient.Create(context.TODO(), &prog)).To(Succeed())
+
+				stack.Spec.ProgramRef = &shared.ProgramReference{
+					Name: prog.Name,
+				}
+
+				// Set DestroyOnFinalize to clean up the configmap for repeat runs.
+				stack.Spec.DestroyOnFinalize = true
+				stack.Name = "changing-program-" + randString()
+				Expect(k8sClient.Create(context.TODO(), &stack)).To(Succeed())
+				waitForStackSuccess(&stack)
+
+				Expect(stack.Status.LastUpdate).ToNot(BeNil())
+				revisionOnFirstSuccess = stack.Status.LastUpdate.LastSuccessfulCommit
+				fmt.Fprintf(GinkgoWriter, ".status.lastUpdate.LastSuccessfulCommit before changing program: %s", revisionOnFirstSuccess)
+				prog2 := programFromFile("testdata/test-program-changed.yaml")
```

Comment:
nit: different style of path from the previous load. I'm pretty sure they're semantically equivalent though.